test/nodes/test_agent_rocketride.py

```
@@ -0,0 +1,404 @@
+
+    def test_write_questions_no_memory_raises(self):
+        """WriteQuestions should raise ValueError if no memory node is connected."""
+        inst = IInstance()
+        inst.IGlobal = MagicMock()
+        inst.IGlobal.agent = MagicMock()
+        inst.instance = MagicMock()
+
+        mock_host = MagicMock()
+        mock_host.memory = None  # No memory connected
+        with patch('nodes.agent_rocketride.IInstance.AgentHostServices', return_value=mock_host):
+            with pytest.raises(ValueError, match='memory'):
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)